From 6e8405bd916cbbec9436f092057ba4f85375b5cf Mon Sep 17 00:00:00 2001
From: Judith Mendez <jm@ti.com>
Date: Mon, 2 Feb 2026 13:59:25 -0600
Subject: [PATCH 3/7] PENDING: mmc: sdhci_am654: Enable DLL based on
 mmc->actual_clock

Switch to mmc->actual_clock for DLL enable decision making. The actual
MMC clock is better to use in cases that controller clock frequency
is not the expected 200MHz. DLL enablement should then be determined
based on actual_clock > 50MHz & mode > MMC_TIMING_UHS_SDR25.

Signed-off-by: Judith Mendez <jm@ti.com>
---
 drivers/mmc/host/sdhci_am654.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/mmc/host/sdhci_am654.c b/drivers/mmc/host/sdhci_am654.c
index 33747ad126b1..21804148426f 100644
--- a/drivers/mmc/host/sdhci_am654.c
+++ b/drivers/mmc/host/sdhci_am654.c
@@ -316,8 +316,8 @@ static void sdhci_am654_set_clock(struct sdhci_host *host, unsigned int clock)
 
 	regmap_update_bits(sdhci_am654->base, PHY_CTRL4, mask, val);
 
-	if (timing > MMC_TIMING_UHS_SDR25 && clock >= CLOCK_TOO_SLOW_HZ) {
-		sdhci_am654_setup_dll(host, clock);
+	if (timing > MMC_TIMING_UHS_SDR25 && host->mmc->actual_clock >= CLOCK_TOO_SLOW_HZ) {
+		sdhci_am654_setup_dll(host, host->mmc->actual_clock);
 		sdhci_am654->dll_enable = true;
 
 		if (timing == MMC_TIMING_MMC_HS400) {
-- 
2.53.0

